content/blog/managing-aws-credentials-on-cicd/index.md

```diff
@@ -0,0 +1,231 @@
+
+<!--more-->
+
+You’ve heard about *Infrastructure as Code* and decided to try your hand at using Pulumi to manage cloud infrastructure using real programming languages. Naturally, you’re curious and eager to start updating your cloud infrastructure in your CI/CD pipeline – but then panic hits. Is updating your cloud infrastructure secure? If you start doing cloud deployments in your CI/CD system, doesn’t that mean you need to give it your production credentials? Aren’t those the most sensitive things your company has?
+
+Take a deep breath. It’s going to be OK. You can securely provide AWS credentials to your CI/CD system and practice continuous delivery. This post covers all the things you need to know to do it.
+
+The goal of this article is to have a clear understanding of how to properly use Pulumi to update AWS infrastructure within a hosted CI/CD service like [CircleCI](https://circleci.com), [GitLab CI](https://about.gitlab.com/product/continuous-integration/), or [Travis CI](https://travis-ci.org).
+
+> **NOTE:** These recommendations do not apply ff you are running your own CI/CD system within your AWS account, e.g., running a Jenkins server on an EC2 instance or using [AWS CodeDeploy](https://aws.amazon.com/codedeploy/). There are established methods to [assume IAM Roles from applications running on an EC2 instance](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2.html). In that case, refer to the documentation specific to your CI/CD system for how to handle AWS credentials.
+
+Depending on your specific environment, needs, and constraints, there may be a better approach for you to take. These recommendations describe a general “one-size fits most” approach that requires a minimal amount of work to configure and maintain.
+
+The approach includes the following:
+
+- Create a dedicated IAM User for your CI/CD
+- Provide the IAM User’s credentials to your CI/CD system
+- Automate rotating and revoking that user’s credentials
+- During deployments, exchange the IAM User credentials for more privileged IAM Role credentials
+
+The process may sound complicated, but we’ll cover each step in detail.  We’ll point out additional ways to enhance your security, such as using Pulumi’s support for client-side encryption to limit access to sensitive information.
+
+## Create a dedicated IAM User for your CI/CD
+
+The first step for securely automating CI/CD on AWS is to create a dedicated IAM User for use in your CI/CD pipelines.
+
+By default, a new IAM User does not have any permissions associated with it, meaning it can only authenticate itself with its access key and secret. Let’s walk through how to grant this IAM User the permissions necessary to update your cloud infrastructure later. For now, we limit this user to have as little access as possible.
```

Comment:
This paragraph is a little confusing. Perhaps rephrase it? Or drop the part about "an IAM User without any permissions cannot do anything"?

I think the main things to emphasize are:

- You obviously want to limit the permissions as narrowly as possible.
- This IAM User will _later_ assume an IAM Role that will have more broad permissions, when it actually goes and performs a deployment.

So in the following code snippet, we're just creating an IAM User that will be used in CI/CD jobs.